### ISSUE TITLE
[Support] HASS 8.5 Annke DVR Integration

### ISSUE THREAD

**Issue by plloppii** [blakeblackshear/frigate-hass-addons#80]
Posting as a last resort as I cannot figure out why my rtsp cameras are not showing up on frigate. 
I am attempting to attach my annke DW81KD DVR camera to frigate through rtsp. 

My frigate.yml
```
mqtt:
  host: 10.1.10.211
  user: mqtt-user
  password: mqtt-pass

cameras:
  frigate_cam1:
    rtmp:
      enabled: false 
    ffmpeg:
      inputs:
        - path: "rtsp://username:password@169.254.62.228:554/Streaming/Channels/202"
          roles:
              - detect
    detect:
      width: 960
      height: 480
      fps: 5

detectors:
  cpu1:
    type: cpu
  cpu2:
    type: cpu
```

Seems simple enough, I can access that rtsp stream through VLC player perfectly fine. 

Frigate logs are as followed:
```
rtsp://homeassistant:password12345@169.254.62.228:554/Streaming/Channels/202: Connection timed out
[2022-08-25 15:12:43] frigate.video                  ERROR   : frigate_cam1: Unable to read frames from ffmpeg process.
[2022-08-25 15:12:43] frigate.video                  ERROR   : frigate_cam1: ffmpeg process is not running. exiting capture thread...
[2022-08-25 15:12:48] watchdog.frigate_cam1          ERROR   : Ffmpeg process crashed unexpectedly for frigate_cam1.
[2022-08-25 15:12:48] watchdog.frigate_cam1          ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-08-25 15:12:48] ffmpeg.frigate_cam1.detect     ERROR   : [tcp @ 0x559aab7a20] Connection to tcp://169.254.62.228:554?timeout=5000000 failed: Connection timed out
```

I'm getting a bunch of connection timeouts. 
I've troubleshooted a ton, attempting to add input_args, and output_args according to [this](https://docs.frigate.video/configuration/camera_specific), but without avail.
 
I attempted to add the stream manually through home assistant and I can get the stream to work ok through a manually configured `generic camera`.
```
stream:
camera:
  - platform: generic
    name: "config_camera"
    stream_source: "rtsp://username:password@169.254.62.228:554/Streaming/Channels/201"
    username: username
    password: password
    authentication: basic
```

Does anyone have any other tips of troubleshooting this?
I see people running the `ffprobe` command, and also `ffmpeg` in the ssh terminal but I'm not sure how to do so in my HASS instance.

**Comment by plloppii** [comment]
Haha I went from a pretty complex user and pass down to something more simple, but yeah I can dumb it down some more. 
I can ping my DVR perfectly fine from my hass instance, so its a bit baffling.

202 is a sub stream and 201 is main stream, and I found both to work with HASS. 
No luck on frigate side of things.

**Comment by NickM-27** [comment]
Let me know if that works, worst case you could try using rtsp-simple-server and pulling the stream with frigate from there instead of the camera itself, but most likely it is a password issue.

**Comment by plloppii** [comment]
Didnt end up working. I'll try out rtsp-simple-server!
I also bought a cheap ip camera coming in to do a sanity check. 
I assume frigate doesnt have a authorization yaml block that I can test out?

**Comment by NickM-27** [comment]
It's all done in the rtsp url. It is just weird, frigate should be able to see camera if HA can unless there is some firewall or other rule keeping it from communicating.

**Comment by plloppii** [comment]
Okay so I figured it out. 

This was my setup:
Annke NVR, eth0(169.254.62.228)
HASS, eth0(169.254.62.228) wlan0(10.1.10.211)
My access computer, wlan0(10.1.10.23)

This morning, I put all devices on the same switch, manually configured eth0 to have 10.0.0.0/24 subnet, 
And frigate started working. 

I'm not understanding something here I guess. 
I thought that the port 80 http access through HASS is all within the scope of the device itself. 
I guess frigate requires all devices to be on the same subnet to work.

**Comment by NickM-27** [comment]
I don't know, sounds more like a HASS addon problem, Frigate running in native docker likely works better for that.
